byzcoin/internal/parse.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/BurntSushi/toml"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/kyber/v3/util/encoding"
+	"go.dedis.ch/onet/v3"
+	"go.dedis.ch/onet/v3/app"
+	"go.dedis.ch/onet/v3/log"
+
+	"go.dedis.ch/onet/v3/network"
+)
+
+// StrToSI converts a string to a ServerIdentity. If tls is true, it will se the URL to point
+// to the address of the string and the port +1.
+func StrToSI(addr string, tls bool) (*network.ServerIdentity, error) {
+	if tls {
+		return AddrToSI(network.NewAddress(network.TLS, addr))
+	}
+	return AddrToSI(network.NewAddress(network.PlainTCP, addr))
+}
+
+// AddrToSI converts a network.Address to a ServerIdentity. If the address is of type TLS,
+// it sets the URL of the ServerIdentity to connect using TLS.
+func AddrToSI(addr network.Address) (*network.ServerIdentity, error) {
+	si := &network.ServerIdentity{Address: addr}
+	if addr.ConnType() == network.TLS {
+		port, err := strconv.Atoi(addr.Port())
+		if err != nil {
+			return nil, err
+		}
+		si.URL = fmt.Sprintf("https://%s:%d", addr.Host(), port+1)
+	}
+	return si, nil
+}
+
+// LoadRoster gets the roster from a file.
+func LoadRoster(file string) (*onet.Roster, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	g, err := app.ReadGroupDescToml(f)
+	if err != nil {
+		return nil, err
+	}
+	return g.Roster, nil
+}
+
+// LoadPrivate loads a private.toml and returns a ServerIdentity.
+func LoadPrivate(file string) (*network.ServerIdentity, error) {
```

Comment:
I don't understand why this exists, when onet/app/config.go's ParseCothority exists.